packages/client/auth-core/src/contexts/authlogin.tsx

```diff
@@ -78,18 +77,23 @@ export function AuthProvider({ children, crossmint, setJwtToken, appearance }: A
         }
     }, [crossmint.jwt]);
 
+    const value = useMemo(() => ({ login, logout, jwt: crossmint.jwt }), [login, logout]);
+
     return (
-        <AuthContext.Provider value={{ login, logout, jwt: crossmint.jwt }}>
+        <AuthContext.Provider value={value}>
             {children}
-            {modalOpen && (
-                <AuthModal
-                    baseUrl={crossmintService.crossmintBaseUrl}
-                    setModalOpen={setModalOpen}
-                    setJwtToken={setJwtToken}
-                    apiKey={crossmint.apiKey}
-                    appearance={appearance}
-                />
-            )}
+            {modalOpen
+                ? createPortal(
```

Comment:
By using a react portal, we can avoid Next.js hydration errors. By doing this, it ensures that the modal is handled separately from the rest of the app's content. This prevents problems that happen when the HTML generated on the server doesn't match what React expects on the client, which caused our hydration errors. The portal lets the modal live in its own place in the DOM, keeping everything in sync between the server and client.

in my opinion, we really should have used a portal from the get go